Classes/Hooks/PageLayoutController.php

```diff
@@ -77,69 +72,90 @@ public function __construct(PageRenderer $pageRenderer, IconFactory $iconFactory
-                 * ->instance([
-                 * 'itemOnClipboardUid' => $pasteItem,
-                 * 'itemOnClipboardTitle' => $pasteTitle,
-                 * 'copyMode' => $clipboard->clipData['normal']['mode'] ?? '',
-                 * ])
-                 * );*/
-
-                $pAddExtOnReadyCode .= '
-                    top.copyFromAnotherPageLinkTemplate = ' . json_encode('<button type="button" class="t3js-paste-new btn btn-default" title="' . $this->helper->getLanguageService()->sL('LLL:EXT:paste_reference/Resources/Private/Language/locallang_db.xml:tx_paste_reference_js.copyfrompage') . '">' . $this->iconFactory->getIcon(
-                            'actions-insert-reference',
-                            Icon::SIZE_SMALL
-                        )->render() . '</button>') . ';';
-            }
+        if (count($jsLines)) {
+            $javaScript = implode("\n", $jsLines);
+            $this->pageRenderer->addJsInlineCode('pasteReference', $javaScript, true, false, true);
         }
+/*
+if (!empty($this->elFromTable)) {
+    \TYPO3\CMS\Extbase\Utility\DebuggerUtility::var_dump(['debug' => [
+        'ExtConf: disableCopyFromPageButton' => ($this->extensionConfiguration['disableCopyFromPageButton'] ?? false),
+        'USER: disableCopyFromPageButton' => ($this->helper->getBackendUser()->uc['disableCopyFromPageButton'] ?? false),
+        '$javaScript' => $javaScript,
+    ]],__METHOD__.':'.__LINE__);
+} else {
+    \TYPO3\CMS\Extbase\Utility\DebuggerUtility::var_dump(['debug' => [
+        '$javaScript' => $javaScript,
+    ]],__METHOD__.':'.__LINE__);
+}
+*/
+        return '';
+    }
+
+    protected function getButtonTemplate(): string
+    {
+        $title = $this->helper->getLanguageService()->sL($this->LLL . ':tx_paste_reference_js.copyfrompage');
+        $icon = $this->iconFactory->getIcon('actions-insert-reference', Icon::SIZE_SMALL)->render();
+        // the CSS-class "t3js-paste-new" does not exist in system extensions
+        return '<button type="button" class="t3js-paste-new btn btn-default" title="' . $title . '" data-code-src="' . __METHOD__ . ':' . __LINE__ . '">' . $icon . '</button>';
+    }
 
-        $this->pageRenderer->addJsInlineCode(
-            'pasterefExtOnReady',
-            $pAddExtOnReadyCode,
-            true,
-            false,
-            true
+    protected function addJavaScriptModuleInstruction(): void
+    {
+        $JavaScriptModuleInstruction = JavaScriptModuleInstruction::create($this->jsScriptName);
+        /** @var TYPO3\CMS\Core\Page\JavaScriptRenderer */
+        $javaScriptRenderer = $this->pageRenderer->getJavaScriptRenderer();
+        $javaScriptRenderer->addJavaScriptModuleInstruction(
+            $JavaScriptModuleInstruction->instance(
+                $this->getJsArgumentsArray()
+            )
         );
+    }
 
-        return '';
+    protected function getJsArgumentsArray(): array
+    {
+        /** int tt_content.uid */
```

Comment:
?

Reply:
it just for info, with @var it's throwing an error, I'd rather remove it then